tests/src/model/pages/volumes-page.ts

```diff
@@ -0,0 +1,95 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Locator, Page } from '@playwright/test';
+import { waitUntil, waitWhile } from '../../utility/wait';
+import { CreateVolumePage } from './create-volume-page';
+import { MainPage } from './main-page';
+import { VolumeDetailsPage } from './volume-details-page';
+
+export class VolumesPage extends MainPage {
+  readonly createVolumeButton: Locator;
+  readonly pruneVolumesButton: Locator;
+  readonly collectUsageDataButton: Locator;
+
+  constructor(page: Page) {
+    super(page, 'volumes');
+    this.createVolumeButton = this.additionalActions.getByRole('button', { name: 'Create' });
+    this.pruneVolumesButton = this.additionalActions.getByRole('button', { name: 'Prune' });
+    this.collectUsageDataButton = this.additionalActions.getByRole('button', { name: 'Collect usage data' });
+  }
+
+  async openCreateVolumePage(volumeName: string): Promise<CreateVolumePage> {
+    const row = await this.getVolumeRowByName(volumeName);
+    if (row !== undefined) {
+      throw Error('Volume is already created');
+    }
+    await this.createVolumeButton.click();
+    return new CreateVolumePage(this.page);
+  }
+
+  async openVolumeDetails(volumeName: string): Promise<VolumeDetailsPage> {
+    const volumeRow = await this.getVolumeRowByName(volumeName);
+    if (volumeRow === undefined) {
+      throw Error(`Volume: ${volumeName} does not exist`);
+    }
+    await volumeRow.getByText(volumeName).click();
```

Comment:
getByText() should contain the exact param set to true, without it by default it checks as a substring.

Reply:
Preference should be always on `getByRole('specific-role', { name: 'accessible name', exact: true})`, as per @cbr7 comment, this would be right way. (This is generally speaking)

Reply:
Thank you @cbr7  and @odockal for the information shared.